backend/server.js

```diff
@@ -23,6 +63,39 @@ app.get('/', (req, res) => {
   res.send('Hello world')
 })
 
+//a post to '/users' saves a new user that registers with name, email and password (password saves crypted)
+app.post('/users', async (req, res) => {
+try {
+  const {name, email, password} = req.body;
+  const user = new User({name, email, password: bcrypt.hashSync(password)});
+  // Must use "await" here, else the try/catch will not work since "save" an asynch function
+  await user.save()
+  res.status(201).json({id: user._id, accessToken: user.accessToken});
+
+} catch (err) {
+  res.status(400).json({message: 'Could not create user', errors: err.errors});
+}
+});
+
+
+//a get to '/secrets' gives the user access to secrets if the user is authenticated (exists)
+app.get('/secrets/', authenticateUser);
+app.get('/secrets/', (req, res) => {
+  console.log('Inside /secrets')
+res.json({secret: 'This is a message only shown if user is logged in and has an accessToken'});
+});
+
+
+
+app.post('/sessions', async (req, res) => {
+  const user = await User.findOne({email: req.body.email});
+  if(user && bcrypt.compareSync(req.body.password, user.password)){
+    res.json({userId: user._id, accessToken: user.accessToken});
+  }else {
+    res.json({notFound: true});
```

Comment:
Since you don't specify a status code (404,400, etc.) your frontend does not notice that the response is not OK.

In your frontend `response.ok` checks if the status code is a 200 or 201.  So I was able to access the `/Secrets` page with the wrong password
<img width="1060" alt="image" src="https://user-images.githubusercontent.com/807352/75826713-6c3c4700-5da8-11ea-94d0-421c041a1410.png">
Although, it did show the error message, and not the secret.